java/vector/src/main/java/org/apache/arrow/vector/BaseNullableFixedWidthVector.java

```diff
@@ -188,6 +188,7 @@ public void reset() {
 
    @Override
    public void clear() {
+      valueCount = 0;
```

Comment:
Is this an behavior change introduced in this PR?

Reply:
Seems correct. But not sure if this (a) doesn't matter or (b) fix a buggy behavior in the old vector.